src/track/track.cpp

```diff
@@ -746,6 +746,42 @@ void Track::removeCue(const CuePointer& pCue) {
     emit(cuesUpdated());
 }
 
+void Track::removeMainCue() {
+    QMutexLocker lock(&m_qMutex);
+    for (CuePointer pCue : m_cuePoints) {
+        if (pCue->getType() == Cue::LOAD) {
+            disconnect(pCue.get(), 0, this, 0);
+            m_cuePoints.removeOne(pCue);
+        }
+    }
+    markDirtyAndUnlock(&lock);
+    emit(cuesUpdated());
+}
+
+void Track::removeHotCues() {
+    QMutexLocker lock(&m_qMutex);
+    for (CuePointer pCue : m_cuePoints) {
+        if (pCue->getType() == Cue::CUE) {
+            disconnect(pCue.get(), 0, this, 0);
+            m_cuePoints.removeOne(pCue);
+        }
+    }
+    markDirtyAndUnlock(&lock);
+    emit(cuesUpdated());
+}
+
+void Track::removeLoopCues() {
+    QMutexLocker lock(&m_qMutex);
+    for (CuePointer pCue : m_cuePoints) {
+        if (pCue->getType() == Cue::LOOP) {
+            disconnect(pCue.get(), 0, this, 0);
+            m_cuePoints.removeOne(pCue);
```

Comment:
@uklotzde any ideas why this is not working too?

Reply:
I don't remember right now how this works with c++, but in Java you cannot remove an element of a list that you're iterating with a for loop. You need to get an iterator, and make a loop with the iterator. 
Since the iterator only has references to next and previous, it doesn't get affected by removing elements from the list.

Reply:
Then why does [clearing hotcues](https://github.com/mixxxdj/mixxx/pull/1427/files#diff-f59a0f669757d180c782c7a8fa0adea0R766) work?

Reply:
This is definitely wrong: "... any non-const function call performed on the QList will render all existing iterators undefined."

Reply:
Use this one:
http://doc.qt.io/archives/qt-4.8/qmutablelistiterator.html

I remember it, because I just removed it from AnalyzerQueue ;)

Reply:
I tried changing this to `for (const CuePointer& pCue : m_cuePoints) {` but it did not solve the issue.

Reply:
I'm guessing there's something else being triggered elsewhere in the code that actually clears the hotcues...